### PR TITLE
Developement frontend

### PR DESCRIPTION
**Pull request by LiviaMaria21**
Login si logout pe frontend.
Am avut 100 de probleme cu intellij, sa nu va speriati ce ii aici

### CODE REVIEW

File: .idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (popiosifcristian) [inline .idea/compiler.xml]: Please delete idea generated files from pull request

File: teamup-client/src/app/common/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            this.router.navigate(['dashboard']);
+        }
+    }
+
+    /**
+     * Log in the user with a username and a password.
+     *
+     * */
+    login() {
+        this.username = this.form.get('username').value.toString();
```
Comment (popiosifcristian) [inline teamup-client/src/app/common/login/login.component.ts]: Value is already a string so you don`t need toString() for that

File: teamup-client/src/app/common/login/login.validators.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import {AbstractControl, ValidationErrors} from '@angular/forms';
+
+export class LoginValidators {
```
Comment (popiosifcristian) [inline teamup-client/src/app/common/login/login.validators.ts]: I don`t think that we have discussed validity on username, are those validators required?

File: teamup-client/src/app/core/services/auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+export class AuthService {
+  constructor(private http: HttpClient, private router: Router) {
+  }
+
+  /**
+   * This function logs in users and handles eventual errors.
+   * @param user that is going to be logged in
+   */
+  public login(user: UserLogin) {
+    console.log(user);
```
Comment (popiosifcristian) [inline teamup-client/src/app/core/services/auth/auth.service.ts]: I don`t think that you should subscribe in here, just store the token and return an observable. Also remove the console logs

File: teamup-server/target/classes/application.properties [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#Jwt generation secret
```
Comment (popiosifcristian) [inline teamup-server/target/classes/application.properties]: Please also remove target files from your pull request.

File: teamup-client/src/app/core/services/user/user.service.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {User} from '../../models/user';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserService {
+  baseUrl = 'http://localhost:1212';
```
Comment (catapopa) [inline teamup-client/src/app/core/services/user/user.service.ts]: Get the baseUrl from congfig

File: .idea/compiler.xml (hunk 1 above)
Comment (catapopa) [reply .idea/compiler.xml]: add this in .gitignore

**Pull request merged**